tests/Unit/Type/data/custom-query-type.php

```diff
@@ -0,0 +1,29 @@
+<?php
+declare(strict_types=1);
+
+namespace My\Test\Extension\Domain\Repository;
+
+use TYPO3\CMS\Extbase\DomainObject\AbstractEntity;
+use TYPO3\CMS\Extbase\Persistence\QueryInterface;
+use function PHPStan\Testing\assertType;
+
+// Extbase naming convention says this model should be called MyModel, but we want to test that it also works if we
+// have a different model name
+class SomeOtherModel extends AbstractEntity {}
+
+class MyModelRepository extends \TYPO3\CMS\Extbase\Persistence\Repository {
+	public function findBySomething()
+	{
+		/** @var QueryInterface<SomeOtherModel> $query */
+		$query = $this->persistenceManager->createQueryForType(SomeOtherModel::class);
+
+		$result = $query->execute();
+		assertType('TYPO3\CMS\Extbase\Persistence\QueryInterface<My\Test\Extension\SomeOtherModel>', $query);
```

Comment:
Namespace must also be changed here and below

Reply:
Fixed